pkg/collector/corechecks/network-devices/cisco-sdwan/payload/cedge_interface.go

```diff
@@ -0,0 +1,126 @@
+	devicemetadata "github.com/DataDog/datadog-agent/pkg/networkdevice/metadata"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// https://github.com/YangModels/yang/blob/9442dda17a9a5f1f0db548512446e3d9ca37a955/vendor/cisco/xe/17131/Cisco-IOS-XE-interfaces-oper.yang#L305
+var cEdgeOperStatusMap = map[string]devicemetadata.IfOperStatus{
+	"if-oper-state-down":             devicemetadata.OperStatusDown,
+	"if-oper-state-invalid":          devicemetadata.OperStatusDown,
+	"if-oper-state-ready":            devicemetadata.OperStatusUp,
+	"if-oper-state-no-pass":          devicemetadata.OperStatusDown,
+	"if-oper-state-test":             devicemetadata.OperStatusTesting,
+	"if-oper-state-unknown":          devicemetadata.OperStatusUnknown,
+	"if-oper-state-dormant":          devicemetadata.OperStatusDormant,
+	"if-oper-state-not-present":      devicemetadata.OperStatusNotPresent,
+	"if-oper-state-lower-layer-down": devicemetadata.OperStatusLowerLayerDown,
+}
+
+// https://github.com/YangModels/yang/blob/9442dda17a9a5f1f0db548512446e3d9ca37a955/vendor/cisco/xe/17131/Cisco-IOS-XE-interfaces-oper.yang#L212
+var cEdgeAdminStatusMap = map[string]devicemetadata.IfAdminStatus{
+	"if-state-unknown": devicemetadata.AdminStatusDown,
+	"if-state-up":      devicemetadata.AdminStatusUp,
+	"if-state-down":    devicemetadata.AdminStatusDown,
+	"if-state-test":    devicemetadata.AdminStatusTesting,
+}
+
+// CEdgeInterface is an implementation of CiscoInterface for cEdge devices
+type CEdgeInterface struct {
+	client.CEdgeInterfaceState
+}
+
+// ID returns an unique interface ID
+func (itf *CEdgeInterface) ID() string {
+	return fmt.Sprintf("%s:%s", itf.VmanageSystemIP, itf.Ifname)
+}
+
+// Index returns the interface index
+func (itf *CEdgeInterface) Index() int {
+	index, err := strconv.Atoi(itf.Ifindex)
+	if err != nil {
+		log.Warnf("Unable to parse cEdge interface %s index %s", itf.Ifname, itf.Ifindex)
+	}
+	return index
+}
+
+// Speed returns the interface speed
+func (itf *CEdgeInterface) Speed() int {
+	speed, err := strconv.Atoi(itf.SpeedMbps)
+	if err != nil {
+		log.Warnf("Unable to parse cEdge interface %s speed %s", itf.Ifname, itf.SpeedMbps)
+	}
+	return speed
+}
+
+// OperStatus returns the interface oper status
+func (itf *CEdgeInterface) OperStatus() devicemetadata.IfOperStatus {
+	return convertOperStatus(cEdgeOperStatusMap, itf.IfOperStatus)
+}
+
+// AdminStatus returns the interface admin status
+func (itf *CEdgeInterface) AdminStatus() devicemetadata.IfAdminStatus {
+	return convertAdminStatus(cEdgeAdminStatusMap, itf.IfAdminStatus)
+}
+
+// Metadata returns the interface metadata
+func (itf *CEdgeInterface) Metadata(namespace string) devicemetadata.InterfaceMetadata {
+	index, err := strconv.Atoi(itf.Ifindex)
+	if err != nil {
+		log.Warnf("Unable to parse cEdge interface %s index %s", itf.Ifname, itf.Ifindex)
+	}
+
+	return devicemetadata.InterfaceMetadata{
+		DeviceID:    fmt.Sprintf("%s:%s", namespace, itf.VmanageSystemIP),
```

Comment:
I don't understand the DeviceID, how does that differ from `itf.IPAddress` or `itf.Ipv6Address`
I would have assumed we would use the IP of the device instead of the vmanage IP.

If that's expected, can you put a small comment explaining it?